### PR TITLE
Modify Unit Tests GitHub Actions workflow to get code coverage test reports

### PR DESCRIPTION
**Pull request by asanzgom**
Modify Unit Tests GitHub Actions workflow to implement codecov in order to get code coverage test reports

### CODE REVIEW

**Comment by asanzgom** [comment]
@VaishnaviHire @zdtsw @ykaliuta @StevenTobin @csams @biswassri Can you please add yourselves as reviewers? I cannot

File: .github/workflows/codecov.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      - name: Install dependencies
+        run: go mod download
+
+      - name: Run tests
+        run: go test -coverprofile=coverage.txt
+
+      - name: Upload results to Codecov
+        uses: codecov/codecov-action@v4
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
```
[the remark this answers is not in the document]
Comment (zdtsw) [reply .github/workflows/codecov.yml:28]: no need explicity set files: coverage.txt ?

**Comment by zdtsw** [comment]
but all in all, why not hook this in to unit-test but create one workflow for it?

File: .github/workflows/codecov.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+          fetch-depth: 0
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+
+      - name: Install dependencies
+        run: go mod download
+
+      - name: Run tests
+        run: go test -coverprofile=coverage.txt
```
[the remark this answers is not in the document]
Comment (asanzgom) [reply .github/workflows/codecov.yml:23]: coverpofile ensures that we collect coverage, and the metrics will be it output to the text file the results

File: .github/workflows/codecov.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    name: Run tests and collect coverage
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
```
[the remark this answers is not in the document]
Comment (asanzgom) [reply .github/workflows/codecov.yml:17]: @zdtsw what is the change here?

**Comment by asanzgom** [comment]
> but all in all, why not hook this in to unit-test but create one workflow for it?

I can do that if you all agree

File: .github/workflows/codecov.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+      - name: Install dependencies
+        run: go mod download
```
[the remark this answers is not in the document]
Comment (asanzgom) [reply .github/workflows/codecov.yml:20]: that is generated by default, maybe not

File: .github/workflows/codecov.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (asanzgom) [reply .github/workflows/codecov.yml:28]: for that part we will need to establish a token with an account and help and from a repository owner, in the meanwhile this is just the workflow that will make use of it

File: .github/workflows/codecov.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (zdtsw) [reply .github/workflows/codecov.yml:23]: maybe i did not make myself clear in that comments.
have you tested this at least locally to see it worked before created this PR or if any runs from GHA can be backed for this change.

[the remark this answers is not in the document]
Comment (asanzgom) [reply .github/workflows/codecov.yml:23]: Yes, I did implement codecov and get a test coverage report on a sample repo:
image

**Comment by ajaypratap003** [comment]
> > but all in all, why not hook this in to unit-test but create one workflow for it?
> 
> I can do that if you all agree

@asanzgom Yes, I think it would be good to hook with unit-test as code coverage is about unit-testing

File: .github/workflows/codecov.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
+name: Run tests and upload coverage
+
+on: 
+  push
```
Comment (ajaypratap003) [inline .github/workflows/codecov.yml:4]: Would it not be good idea to have code coverage workflow on PR to see code coverage is decrease on increase after new changes on PR ?

Comment (asanzgom) [reply .github/workflows/codecov.yml:4]: sure @ajaypratap003 adding..

**Comment by asanzgom** [comment]
> > > but all in all, why not hook this in to unit-test but create one workflow for it?
> > 
> > 
> > I can do that if you all agree
> 
> @asanzgom Yes, I think it would be good to hook with unit-test as code coverage is about unit-testing

Consolidated into the unit tests workflow

**Comment by AjayJagan** [comment]
lgtm

**Review by ajaypratap003 — COMMENTED** [review]
LGTM. 
Make sure `CODECOV_TOKEN` token must be set in repo before merging this PR.

**Comment by asanzgom** [comment]
@ajaypratap003 @AjayJagan LGTM label needs to be added

**Review by ajaypratap003 — COMMENTED** [review]
lgtm

**Review by biswassri — APPROVED** [review]
Lgtm

**Pull request merged**